pkg/costmodel/metrics.go

```diff
@@ -474,7 +480,7 @@ func (cmme *CostModelMetricsEmitter) Start() bool {
 				}
 				cmme.LBCostRecorder.WithLabelValues(ingressIP, namespace, serviceName).Set(lb.Cost)
 
-				labelKey := getKeyFromLabelStrings(namespace, serviceName)
+				labelKey := getKeyFromLabelStrings(ingressIP, namespace, serviceName)
```

Comment:
Without `ingressIP`, the key doesn't match the metrics actual labels, so we repeatedly fail to delete the metric.